jfrog-cli/artifactory/commands/buildinfo/addgit.go

```diff
@@ -1,42 +1,312 @@
+	if err != nil {
+		return err
+	}
+
+	// Done.
+	log.Info("Collected git revision and remote url for", config.BuildName+"/"+config.BuildNumber+".")
+	return nil
+}
+
+func (config *BuildAddGitConfiguration) collectBuildIssues() ([]buildinfo.AffectedIssue, error) {
+	log.Info("Collecting build issues from VCS...")
+
+	// Initialize issues-configuration.
+	config.IssuesConfig = new(IssuesConfiguration)
+
+	// Build config from file.
+	err := config.IssuesConfig.populateIssuesConfigurations(config.ConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Run issues collection.
+	return config.doCollect(config.IssuesConfig)
+}
+
+func (config *BuildAddGitConfiguration) doCollect(issuesConfig *IssuesConfiguration) ([]buildinfo.AffectedIssue, error) {
+	// Create regex pattern.
+	issueRegexp, err := clientutils.GetRegExp(issuesConfig.Regexp)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create services manager to get build-info from Artifactory.
+	sm, err := utils.CreateServiceManager(issuesConfig.ArtDetails, false)
+	if err != nil {
+		return nil, err
+	}
+	// Get latest build-info from Artifactory.
+	buildInfoParams := services.BuildInfoParams{BuildName: config.BuildName, BuildNumber: config.BuildNumber}
+	buildInfo, err := sm.GetBuildInfo(buildInfoParams)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get previous VCS Revision from BuildInfo.
+	lastVcsRevision := ""
+	if buildInfo.Vcs != nil {
+		lastVcsRevision = buildInfo.Vcs.Revision
+	}
+
+	// Get log with limit, from latest commit.
+	logCmd := &LogCmd{gitPath: config.DotGitPath, logLimit: issuesConfig.LogLimit, lastVcsRevision: lastVcsRevision}
+	var foundIssues []buildinfo.AffectedIssue
+	protocolRegExp := gofrogcmd.CmdOutputPattern{
+		RegExp: issueRegexp,
+		ExecFunc: func(pattern *gofrogcmd.CmdOutputPattern) (string, error) {
+			// Reached here - means no error occurred.
+
+			// Check for out of bound results.
+			if len(pattern.MatchedResults) - 1 < issuesConfig.KeyGroupIndex || len(pattern.MatchedResults) - 1 < issuesConfig.SummaryGroupIndex {
+				return "", errors.New("Unexpected result while parsing build-issues from git log")
+			}
+			// Create found Affected Issue.
+			foundIssue := buildinfo.AffectedIssue{Key: pattern.MatchedResults[issuesConfig.KeyGroupIndex], Summary: pattern.MatchedResults[issuesConfig.SummaryGroupIndex], Aggregated: false}
+			if issuesConfig.TrackerUrl != "" {
+				foundIssue.Url = issuesConfig.TrackerUrl + pattern.MatchedResults[issuesConfig.KeyGroupIndex]
+			}
+			foundIssues = append(foundIssues, foundIssue)
+			log.Debug("Found issue: " + pattern.MatchedResults[issuesConfig.KeyGroupIndex])
+			return "", nil
+		},
+	}
+
+	// Change working dir to where .git is.
+	os.Chdir(config.DotGitPath)
+
+	// Run git command.
+	_, exitOk, err := gofrogcmd.RunCmdWithOutputParser(logCmd, false, &protocolRegExp)
+	if err != nil || !exitOk {
+		// Error occurred during git log execution. Happens in any case git prints to stderr.
```

Comment:
I'm not sure that the above comment is always correct. Maybe we shoild remove it.